src/main/java/org/mule/module/apikit/validation/attributes/QueryParameterValidator.java

```diff
@@ -52,7 +52,9 @@ private static ValidatedQueryParams validateQueryParams(Map<String, Parameter> q
       List<String> values = incomingQueryParams.getAll(paramKey);
 
       if (!values.isEmpty()) {
-        if (parameterDefinition.isRepeat() || parameterDefinition.isArray()) {
+        if (isNull(values) && parameterDefinition.isNullable()) {
+          validateQueryParam(queryParamsCopy, paramKey, parameterDefinition, values.get(0));
```

Comment:
`isNull(null)` currently evaluates to `true`. That will cause a `NullPointerException` if we try to run `values.get(0)`